packages/app/src/resource/ProfilesPage.tsx

```diff
@@ -0,0 +1,184 @@
+import { OperationOutcome, Resource, ResourceType } from '@medplum/fhirtypes';
+import React, { useCallback, useEffect, useState } from 'react';
+import {
+  Document,
+  SupportedProfileStructureDefinition,
+  ResourceForm,
+  isSupportedProfileStructureDefinition,
+  useMedplum,
+} from '@medplum/react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { showNotification } from '@mantine/notifications';
+import { deepClone, normalizeErrorString, normalizeOperationOutcome } from '@medplum/core';
+import { Button, Code, Group, Stack, Tabs, ThemeIcon } from '@mantine/core';
+import { addProfileToResource, cleanResource } from './utils';
+import { IconCheck } from '@tabler/icons-react';
+
+export function ProfilesPage(): JSX.Element | null {
+  const medplum = useMedplum();
+  const { resourceType, id } = useParams() as { resourceType: ResourceType; id: string };
+  const [resource, setResource] = useState<Resource | undefined>();
+  const [currentProfile, setCurrentProfile] = useState<SupportedProfileStructureDefinition>();
+  const [availableProfiles, setAvailableProfiles] = useState<SupportedProfileStructureDefinition[]>();
+
+  useEffect(() => {
+    medplum
+      .readResource(resourceType, id)
+      .then((resource) => setResource(deepClone(resource)))
+      .catch((err) => {
+        showNotification({ color: 'red', message: normalizeErrorString(err) });
+      });
+  }, [medplum, resourceType, id]);
+
+  // This is a bit inefficient since the entire structure definition
+  // for each available profile is being fetched. All that is really needed is the title & url
+  // The SD is useful for the time being to populate the Snapshot and JSON debugging tabs;
+  // but those will likely be removed before deploying
+  useEffect(() => {
+    medplum
+      .searchResources('StructureDefinition', { type: resourceType, derivation: 'constraint' })
+      .then((results) => {
+        const supported = results.filter(isSupportedProfileStructureDefinition);
+        setAvailableProfiles(supported);
+      })
+      .catch(console.error);
+  }, [medplum, resourceType]);
+
+  useEffect(() => {
+    if (availableProfiles) {
+      const activeProfiles = availableProfiles.filter(
+        (profile) => resource?.meta?.profile?.includes(profile.url) ?? false
+      );
+      if (activeProfiles[0]) {
+        setCurrentProfile(activeProfiles[0]);
+      }
+    }
+  }, [resource, availableProfiles]);
+
+  if (!resource) {
+    return null;
+  }
+
+  return (
+    <Document>
+      <h2>Available {resourceType} profiles</h2>
```

Comment:
Minor:  We've been trying to use Mantine's `<Title order={2}>` rather than `<h2>`